web-server/v0.4/src/models/search.js

```diff
@@ -53,11 +53,12 @@ export default {
         let searchResults = {};
         searchResults['resultCount'] = response.hits.total;
         let parsedResults = [];
-        response.hits.hits.map(result => {
+        response.hits.hits.map((result, index) => {
           let parsedResult = {};
           selectedFields.map(field => {
             parsedResult[field] = result._source[field.split('.')[0]][field.split('.')[1]];
           });
+          parsedResult['key'] = index;
```

Comment:
Can you assert that we have an actual string for the index name here?

Reply:
I've updated the key assignment to take the result id if the field exists. Since we have stable IDs for rendering items, we can remove the index field referenced by the `map()` method.